charts/tidb-cluster/templates/scripts/_start_tidb.sh.tpl

```diff
@@ -26,7 +26,12 @@ then
     tail -f /dev/null
 fi
 
+# Use HOSTNAME if POD_NAME is unset for backward compatibility.
+POD_NAME=${POD_NAME:-$HOSTNAME}
```

Comment:
This line should also be protected by trigger

Reply:
this is fine because it works before and after the change, and it does not affect StatefulSet spec.

the new NAMESPACE/HEADLESS_SERVICE_NAME/POD_NAME envs should only be set on TiDB statefulset only when enableAdvertiseAddress is enabled.